clients/ui/frontend/src/app/AppRoutes.tsx

```diff
@@ -20,23 +22,52 @@ export const useAdminSettings = (): NavDataItem[] => {
   ];
 };
 
-export const useNavData = (): NavDataItem[] => [
-  {
-    label: 'Model Registry',
-    path: '/model-registry',
-  },
-  ...useAdminSettings(),
-];
+export const useNavData = (): NavDataItem[] => {
+  const { config } = useModularArchContext();
+  const { deploymentMode } = config;
+  const isStandalone = deploymentMode === DeploymentMode.Standalone;
+  const isFederated = deploymentMode === DeploymentMode.Federated;
+
+  const baseNavItems = [
+    {
+      label: 'Model Registry',
+      path: '/model-registry',
+    },
+  ];
+
+  // Only show Model Catalog in Standalone or Federated mode
+  if (isStandalone || isFederated) {
+    baseNavItems.push({
+      label: 'Model Catalog',
+      path: '/model-catalog',
+    });
+  }
```

Comment:
@lucferbux is this sufficient for now to hide the incomplete model catalog feature from normal kubeflow installations or do we need some kind of feature flag solution?

Reply:
We would need a feature flag solution soon enough  upstream cc @christianvogt (we can start talking aboud adding that in our library) but this should be enought from now.